galaxykit/containerutils.py

```diff
@@ -37,11 +41,21 @@ def __init__(
         self.engine = engine
         self.registry = registry
         self.tls_verify = tls_verify
+        self.auth = auth
 
-        if auth:  # we only need to auth if creds are supplied
+        if auth: # we only need to auth if creds are supplied
             self.login(*auth, fail_ok=True)
+    
+    def _check_login(self):
+        if self.auth:
```

Comment:
is it possible to reach this point without having been authenticated ? I think an exception is raised if no auth is provided when initializing the container client so maybe we don't need this conditional

Reply:
Yes, because the client can be used in an unauthenticated mode.